.chloggen/1006.yaml

```diff
@@ -0,0 +1,13 @@
+change_type: breaking
+
+component: messaging
+
+note: |
+  Support generic operations in messaging and rename metrics:
+
+    - Make `messaging.operation.name` required and `messaging.operation.type` conditionally required when type is applicable.
+    - Rename `messaging.publish.messages` metric to `messaging.client.published.messages`
+    - Unify `messaging.publish.duration` and `messaging.receive.duration` metrics into `messaging.client.request.duration`
+    - Unify `messaging.receive.messages` and `messaging.process.messages` metrics into `messaging.client.consumed.messages`
```

Comment:
Shouldn't these also be added to the schema change (renamed)?

Reply:
good question. It's a rename with two new attributes, so we don't have a way to transform old -> new (`messaging.operation.name` was not available in any shape on the old metric).